libraries/aws/provisioning/test/system/aws_iot_tests_provisioning_system.c

```diff
@@ -179,11 +179,66 @@ static void _printKeysAndCertificateCallback( void * contextParam,
 
 /*-----------------------------------------------------------*/
 
-static void _storeCertificateDataForProvisioning( void * contextParam,
-                                                  const AwsIotProvisioningCreateKeysAndCertificateResponse_t * pResponseInfo )
+/**
+ * @brief User callback function for printing parsed response data from the
+ * MQTT CreateCertificateFromCsr service API.
+ */
+static void _printCertificateFromCsrCallback( void * contextParam,
+                                              const AwsIotProvisioningCreateCertFromCsrResponse_t * pResponseInfo )
 {
-    _keysAndCertificateCallbackContext_t * certificateIdTokenContext =
-        ( _keysAndCertificateCallbackContext_t * ) contextParam;
+    ( void ) contextParam;
+    TEST_ASSERT_NOT_NULL( pResponseInfo );
+
+    IotLogInfo( "\n Status Code = %d\n", pResponseInfo->statusCode );
+
+    if( pResponseInfo->statusCode == AWS_IOT_PROVISIONING_SERVER_STATUS_ACCEPTED )
+    {
+        /* Check parameters against expected values. */
+        TEST_ASSERT_NOT_NULL( pResponseInfo->u.acceptedResponse.pDeviceCert );
+        TEST_ASSERT_GREATER_THAN( 0, pResponseInfo->u.acceptedResponse.deviceCertLength );
+
+        IotLogInfo( "\n Certificate PEM = %.*s\n Certificate ID = %.*s\n Ownership Token = %.*s\n",
+                    pResponseInfo->u.acceptedResponse.deviceCertLength,
+                    pResponseInfo->u.acceptedResponse.pDeviceCert,
+                    pResponseInfo->u.acceptedResponse.certIdLength,
+                    pResponseInfo->u.acceptedResponse.pCertId,
+                    pResponseInfo->u.acceptedResponse.ownershipTokenLength,
+                    pResponseInfo->u.acceptedResponse.pCertOwnershipToken );
+    }
+    else
+    {
+        _printRejectedResponse( &pResponseInfo->u.rejectedResponse );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Callback function that tests for the "Invalid CSR" status in the
+ * parsed server response in the failure case of calling of the CreateCertFromCsr API
+ * with an invalid CSR string.
+ */
+static void _checkInvalidCsrServerResponseCallback( void * contextParam,
+                                                    const AwsIotProvisioningCreateCertFromCsrResponse_t * pResponseInfo )
+{
+    ( void ) contextParam;
+    TEST_ASSERT_NOT_NULL( pResponseInfo );
+
+    /* Make sure that we have received an "Invalid CSR" response from the server. */
+    TEST_ASSERT_MESSAGE( pResponseInfo->statusCode == AWS_IOT_PROVISIONING_SERVER_STATUS_INVALID_CSR,
+                         "Callback didn't receive expected server response status: "
+                         "ExpectedResponseStatus={InvalidCsr}" );
+
+    _printRejectedResponse( &pResponseInfo->u.rejectedResponse );
+}
+
+/*-----------------------------------------------------------*/
+
+static void _storeDataFromKeysAndCertResponseCallback( void * contextParam,
```

Comment:
nitpick: since you documented the other static functions, I may be helpful to document these ones too.

Reply:
Added